src/main/java/mate/academy/bookshop/validation/FieldMatchValidator.java

```diff
@@ -0,0 +1,39 @@
+package mate.academy.bookshop.validation;
+
+import jakarta.validation.ConstraintValidator;
+import jakarta.validation.ConstraintValidatorContext;
+import java.lang.reflect.Field;
+import java.util.Arrays;
+
+public class FieldMatchValidator implements ConstraintValidator<FieldMatch, Object> {
+    private String firstField;
+    private String secondField;
+
+    @Override
+    public void initialize(FieldMatch constraintAnnotation) {
+        firstField = constraintAnnotation.firstField();
+        secondField = constraintAnnotation.secondField();
+    }
+
+    @Override
+    public boolean isValid(Object object, ConstraintValidatorContext constraintValidatorContext) {
+        Object firstFieldValue = getFieldValueByName(object, firstField);
+        Object secondFieldValue = getFieldValueByName(object, secondField);
+
+        return firstFieldValue != null && firstFieldValue.equals(secondFieldValue);
+    }
+
+    private Object getFieldValueByName(Object object, String fieldName) {
+        Field[] fields = object.getClass().getDeclaredFields();
+        Field field = Arrays.stream(fields)
+                .filter(f -> f.getName().equals(fieldName))
+                .findFirst()
+                .orElseThrow(() -> new IllegalArgumentException("Field with this name not found"));

```

Comment:
I think there are already existing method that can take field by name, please use it